### ISSUE TITLE
Fix failing locale tests on Apple platforms with the RU locale

### ISSUE THREAD

**Issue by ldionne** [llvm/llvm-project#45084]
|  |  |
| --- | --- |
| Bugzilla Link | [45739](https://llvm.org/bz45739) |
| Version | unspecified |
| OS | All |
| CC | @arichardson,@mclow |

## Extended Description 
The following tests are failing on Apple platforms:

    test/std/localization/locale.categories/category.monetary/locale.money.get/locale.money.get.members/get_long_double_ru_RU.pass.cpp b/libcxx/test/std/localization/locale.categories/category.monetary/locale.money.get/locale.money.get.members/get_long_double_ru_RU.pass.cpp
    test/std/localization/locale.categories/category.monetary/locale.money.put/locale.money.put.members/put_long_double_ru_RU.pass.cpp b/libcxx/test/std/localization/locale.categories/category.monetary/locale.money.put/locale.money.put.members/put_long_double_ru_RU.pass.cpp

We should figure out why and fix them.

**Comment by mstorsjo** [comment]
I ran into this issue, while poking those test files, and I went ahead and tried to fix it - see https://reviews.llvm.org/D120316.

I also agree that we should remove the trailing space there; that matches other similar tests. This test seemed to be unusually untested in the sense that it was XFAILed in essentially all configurations in our current CI except for Apple platforms, and on Apple platforms, these bits were ifdeffed out.

On top of that, I made https://reviews.llvm.org/D120317 which makes the test a bit more flexible, so that it passes on Glibc, FreeBSD and Windows too.
